Assets/Scripts/PlayerController.cs

```diff
@@ -0,0 +1,92 @@
+using UnityEngine;
+
+public class PlayerController : MonoBehaviour
+{
+    private Animator animator;
+    private Rigidbody2D rb;
+    
+    [SerializeField] private float moveSpeed = 5f;
+    [SerializeField] private float jumpForce = 10f;
+    [SerializeField] private LayerMask groundLayer;
+    [SerializeField] private Transform groundCheck;
+    [SerializeField] private float groundCheckRadius = 0.1f;
+
+    private bool isGrounded;
+    private float horizontalMove = 0f;
+    private bool isJumping = false;
+
+    void Start()
+    {
+        animator = GetComponent<Animator>();
+        rb = GetComponent<Rigidbody2D>();
+        // to prevent rotation of player
+        rb.freezeRotation = true; 
+    }
+
+    void Update()
+    {
+        // checking if player is on the ground and if they are trying to move or jump
+        isGrounded = Physics2D.OverlapCircle(groundCheck.position, groundCheckRadius, groundLayer);
+        horizontalMove = Input.GetAxisRaw("Horizontal") * moveSpeed;
+
+        if (Input.GetKeyDown(KeyCode.UpArrow) && isGrounded)
+        {
+            isJumping = true;
+        }
+
+        UpdateAnimationState();
+    }
+
+    void FixedUpdate()
+    {
+        rb.velocity = new Vector2(horizontalMove, rb.velocity.y);
+
+        if (isJumping)
+        {
+            rb.AddForce(new Vector2(0f, jumpForce), ForceMode2D.Impulse);
+            isJumping = false;
+        }
+    }
+
+    void UpdateAnimationState()
+    {
+        // updating animation of character based on action
+        if (isGrounded)
+        {
+            if (Mathf.Abs(horizontalMove) > 0.1f)
```

Comment:
kind of insignificant but maybe you can take isJumpUp and isJumpDown out like how you did for isRunning and isIdle in the else statement for consistency